ocaml/xapi/xapi_vdi_helpers.ml

```diff
@@ -23,6 +23,19 @@ open Threadext
 module D=Debug.Make(struct let name="xapi" end)
 open D
 
+let attach_rrd_vdi ~__context ~sr=
+	let vdis = Db.SR.get_VDIs ~__context ~self:sr in
+	let vdi =
+		if (List.exists (fun vdi -> (Db.VDI.get_type ~__context ~self:vdi) = `rrd) vdis) then
+			List.hd (List.filter (fun vdi -> Db.VDI.get_type ~__context ~self:vdi = `rrd) vdis)
```

Comment:
This will be much more efficient if you use `Db.SR.get_refs_where` and filter on the SR ref and the VDI type.

The code would look something like [this](https://github.com/xapi-project/xen-api/blob/ec69fe9616c8a13c16b0b0e69f27f8854f7a402b/ocaml/xapi/xapi_pool.ml#L201-204).